tremor-script/src/interpreter.rs

```diff
@@ -388,8 +388,9 @@ where
     'script: 'event,
     'event: 'run,
 {
-    let mut subrange: Option<(usize, usize)> = None;
-    let mut current: &Value = match path {
+    // Fetch the base of the path
+    // TODO: Extract this into a method on `Path`?
```

Comment:
👍 seems sensible

Reply:
It won't be so straightforward to make this abstract enough that both `resolve` and `present` can use it, since in case of `present` sometimes `Cow::Borrowed(&FALSE)` has to be returned in cases where `resolve` would return an `Err(_)`. That `Cow` makes sense for neither the `Ok(_)`, nor the `Err(_)` branch.

I see one architecturally clean option for deduplicating code here and lower in the functions `resolve` and `present` (and perhaps `assign`) but that solution is way out of scope for this PR: if the interpreter's errors were restructured such that all `ErrorKind`s which have to do with the non-presence of a value are grouped into a new variant `ErrorKind::NotPresent(NotPresentError)`, then the abstracting methods could behave ± like the code in `resolve`, and `present` could simply

```rust
match res {
  Ok(res) => res,
  Err(ErrorKind::NotPresent(_np)) => return Ok(Cow::Borrowed(&FALSE)),
  Err(other) => return Err(other)
}
```

In any case: I won't handle this `TODO` in this PR.